ownership-chain/pallets/laos-evolution/src/lib.rs

```diff
@@ -194,6 +211,36 @@ impl<T: Config> EvolutionCollection<AccountIdOf<T>, TokenUriOf<T>> for Pallet<T>
 
 		Ok(())
 	}
+
+	fn enable_public_minting(who: AccountIdOf<T>, collection_id: CollectionId) -> DispatchResult {
+		ensure!(
+			CollectionOwner::<T>::contains_key(collection_id),
+			Error::<T>::CollectionDoesNotExist
+		);
+		ensure!(CollectionOwner::<T>::get(collection_id) == Some(who), Error::<T>::NoPermission);
```

Comment:
what about extracting `is_owner` logic to function, we seem to be using it in a lot of places?